File: c/buffer-overflow.c
#include <stdio.h>
#include <string.h>

#define SIZE 10

void buffer_overflow() {
    char buffer[SIZE];

    for (int i = 0; i <= SIZE; i++) {  // Out-of-bounds access!
        buffer[i] = 'A';  
    }

    printf("Buffer Overflow executed\n");
}

int main() {
    buffer_overflow();
    return 0;
}

File: rust/buffer-overflow.rs
const SIZE: usize = 10;

fn buffer_overflow() {
    let mut buffer = [0u8; SIZE];

    for i in 0..=SIZE {  // This will cause a compilation error!
        buffer[i] = b'A';
    }

    println!("Buffer Overflow executed");
}

fn main() {
    buffer_overflow();
}